src/PowerShell/Microsoft.WinGet.Client/Commands/RepairCommand.cs

```diff
@@ -0,0 +1,165 @@
+        /// </summary>
+        [Parameter(ValueFromPipelineByPropertyName = true)]
+        public SwitchParameter IncludePreRelease { get; set; }
+
+        /// <summary>
+        /// Attempts to repair winget.
+        /// TODO: consider WhatIf and Confirm options.
+        /// </summary>
+        protected override void ProcessRecord()
+        {
+            RepairResult repairResult = RepairResult.Failure;
+
+            var integrityCategory = WinGetIntegrity.GetIntegrityCategory(this);
+
+            this.WriteDebug($"Integrity category type: {integrityCategory}");
+
+            bool preRelease = this.IncludePreRelease.ToBool();
+
+            if (integrityCategory == IntegrityCategory.Installed)
+            {
+                repairResult = this.RepairForInstalled(preRelease, this.Version);
+            }
+            else if (integrityCategory == IntegrityCategory.NotInPath)
+            {
+                this.RepairEnvPath();
+                repairResult = RepairResult.PathUpdated;
+            }
+            else if (integrityCategory == IntegrityCategory.AppInstallerNotRegistered)
+            {
+                var appxModule = new AppxModuleHelper(this);
+                appxModule.RegisterAppInstaller();
+
+                this.WriteDebug($"WinGet version {WinGetVersionHelper.InstalledWinGetVersion} registered");
+                repairResult = RepairResult.Registered;
+            }
+            else if (integrityCategory == IntegrityCategory.AppInstallerNotInstalled ||
+                     integrityCategory == IntegrityCategory.AppInstallerNotSupported ||
+                     integrityCategory == IntegrityCategory.Failure)
+            {
+                if (this.DownloadAndInstall(preRelease, this.Version, false))
+                {
+                    repairResult = RepairResult.Installed;
+                }
+            }
+            else if (integrityCategory == IntegrityCategory.AppExecutionAliasDisabled)
+            {
+                // Sorry, but the user has to manually enabled it.
+                this.WriteInformation(Resources.AppExecutionAliasDisabledHelpMessage, WriteInformationTags);
+                repairResult = RepairResult.NeedsManualRepair;
+            }
+            else
+            {
+                this.WriteInformation(Resources.WinGetNotSupportedMessage, WriteInformationTags);
+            }
+
+            this.WriteObject(repairResult);
+        }
+
+        private RepairResult RepairForInstalled(bool preRelease, string toInstallVersion)
+        {
+            RepairResult repairResult = RepairResult.Failure;
+
+            if (string.IsNullOrEmpty(toInstallVersion))
+            {
+                var gitHubRelease = new GitHubRelease();
+                toInstallVersion = gitHubRelease.GetLatestVersionTagName(preRelease);
+            }
+
+            if (toInstallVersion != WinGetVersionHelper.InstalledWinGetVersion)

```

Comment:
I would love to see implied wildcarding for version matching. For example - If the user specified `-Version 1.3` or `-Version 1.3.2` it would map to the version tag [1.3.2691](https://github.com/microsoft/winget-cli/releases/tag/v1.3.2691). But if they used `-Version 1.3.1` it would map to the version tag [1.3.1872](https://github.com/microsoft/winget-cli/releases/tag/v1.3.1872) and require the `-Prerelease` switch. Whereas `-Version 1.3.20` would map to [1.3.2091](https://github.com/microsoft/winget-cli/releases/tag/v1.3.2091). 

This follows the idea that the cli version is often referenced as the "1.3 Client" or the "1.4 Client", so it's very common to only use the prefix of the version number.

(Yes, I know, I'm asking for scope creep here; Sorry @denelon)